fastddsspy_tool/test/application/CMakeLists.txt

```diff
@@ -0,0 +1,200 @@
+
+    message(STATUS "Finding target for libraries for ${PROJECT_DEPENDENCY}")
+
+    if(TARGET ${PROJECT_DEPENDENCY} OR TARGET eprosima::${PROJECT_DEPENDENCY} OR TARGET ${PROJECT_DEPENDENCY}::${PROJECT_DEPENDENCY})
+        set(TEST_ENVIRONMENT
+            "${TEST_ENVIRONMENT};$<TARGET_FILE_DIR:${PROJECT_DEPENDENCY}>")
+
+    elseif(EXISTS ${PROJECT_DEPENDENCY}_LIBRARY)
+        get_filename_component(
+            ${PROJECT_DEPENDENCY}_LIBRARY_DIR ${${PROJECT_DEPENDENCY}_LIBRARY} DIRECTORY)
+        set(TEST_ENVIRONMENT
+            "${TEST_ENVIRONMENT};${${PROJECT_DEPENDENCY}_LIBRARY_DIR}")
+        unset(${PROJECT_DEPENDENCY}_LIBRARY_DIR)
+
+    else()
+
+        message(STATUS "${PROJECT_DEPENDENCY} could not be added to TEST_ENVIRONMENT")
+
+    endif()
+
+endforeach()
+
+if(WIN32)
+
+    if(TARGET tinyxml2 OR TARGET tinyxml2::tinyxml2)
+        set(TEST_ENVIRONMENT
+            "${TEST_ENVIRONMENT};$<TARGET_FILE_DIR:${TINYXML2_LIBRARY}>")
+    elseif(EXISTS TINYXML2_LIBRARY)
+        get_filename_component(
+            TINYXML2_LIBRARY_DIR ${TINYXML2_LIBRARY} DIRECTORY)
+        set(TEST_ENVIRONMENT
+            "${TEST_ENVIRONMENT};${TINYXML2_LIBRARY_DIR}")
+        unset(TINYXML2_LIBRARY_DIR)
+    endif()
+
+    if(TARGET yamlcpp OR TARGET yamlcpp::yamlcpp)
+        set(TEST_ENVIRONMENT
+            "${TEST_ENVIRONMENT};$<TARGET_FILE_DIR:${YAMLCPP_LIBRARY}>")
+    elseif(EXISTS YAMLCPP_LIBRARY)
+        get_filename_component(
+            YAMLCPP_LIBRARY_DIR ${YAMLCPP_LIBRARY} DIRECTORY)
+        set(TEST_ENVIRONMENT
+            "${TEST_ENVIRONMENT};${YAMLCPP_LIBRARY_DIR}")
+        unset(YAMLCPP_LIBRARY_DIR)
+    endif()
+
+    string(REPLACE ";" "\\;" TEST_ENVIRONMENT "${TEST_ENVIRONMENT}")
+
+endif(WIN32)
+
+set(TEST_NEEDED_SOURCES
+        configuration/configuration_basic.yaml
+        configuration/configuration_wrong_empty_arg.yaml
+        configuration/configuration_wrong_type.yaml
+        configuration/configuration_discovery_time.yaml
+    )
+
+foreach(NEEDED_SOURCE ${TEST_NEEDED_SOURCES})
+    configure_file(${CMAKE_CURRENT_SOURCE_DIR}/${NEEDED_SOURCE}
+        ${CMAKE_CURRENT_BINARY_DIR}/${NEEDED_SOURCE}
+        COPYONLY)
+endforeach()
+
+# populate the tests
+foreach(TEST IN LISTS TEST_LIST)
+    set(TEST_NAME "tool.application.fastddsspy.test.${TEST}")
+
+    if(WIN32)
+        add_test(
+            NAME ${TEST_NAME}
+            COMMAND powershell "-File" ${PWS_LAUNCHER}
+                    ${PYTHON_EXECUTABLE}
+                    ${CMAKE_CURRENT_SOURCE_DIR}/test.py
+                    $<TARGET_FILE:fastddsspy_tool>
+                    ${TEST}
+        )
+
+    else()
+
+        add_test(
+                NAME ${TEST_NAME}
+                COMMAND ${PYTHON_EXECUTABLE}
+                        ${CMAKE_CURRENT_SOURCE_DIR}/test.py
+                        "--exe" $<TARGET_FILE:fastddsspy_tool>
+                        "--test" ${TEST}
+            )
+
+    endif()
+
+    # Add labels to tests
```

Comment:
Add a TODO comment so this should be done implicitly using a cmake_utils function